File: lab05/Right.java
// File: Right.java
// Author: Dr. Watts
// Contents: This file contains the description and implementation
// of a class called Right. 

import static java.lang.Math.*;

public final class Right extends Triangle
{
	private double side2;

	public Right ()
	{
		side2 = 0.0;
	}
	
	public Right (Right R)
	{
		side = R.side;
		side2 = R.side2;
	}
	
	public Right (double S1, double S2)
	{
		side = S1;
		side2 = S2;
	}
	
	public void setSide1 (double S1)
	{
		side = S1;
	}
	
	public double getSide1 ()
	{
		return side;
	}
	
	public void setSide2 (double S2)
	{
		side2 = S2;
	}
	
	public double getSide2 ()
	{
		return side2;
	}
	
	public double perimeter ()
	{
		return side + side2 + sqrt (side * side + side2 * side2);
	}
	
	public double area ()
	{
		return side * side2 / 2;
	}
	
	public String getName ()
	{
		return "Right";
	}

	public static void main (String [] args)
	{
		Right T = new Right ();
		T.write (); System.out.println ();
		T = new Right (7, 5);
		T.write (); System.out.println ();
		Right T2 = new Right (T);
		T2.write (); System.out.println ();
		T2.setSide1 (4.1);
		T2.setSide2 (1.3);
		T2.write (); System.out.println ();
	}
}	
	

File: lab04/Right.cpp
// File: Right.h
// Author: Amandeep Gill
// Contents: the implementation of the Right class

#include <cmath>
#include "Right.h"

using namespace std;

Right::Right()
{
    debug << "Entering default constructor for Right\n";
    side = 0.0;
    side2 = 0.0;
    debug << "Exiting default constructor for Right\n";
}

Right::Right(float S1, float S2)
{
    debug << "Entering second constructor for Right\n";
    side = S1;
    side2 = S2;
    debug << "Exiting second constructor for Right\n";
}

float Right::GetSide1() const
{
    debug << "Entering GetSide1 function for Right\n";
    debug << "Exiting GetSide1 function for Right\n";
    return side;
}

float Right::GetSide2() const
{
    debug << "Entering GetSide2 function for Right\n";
    debug << "Exiting GetSide2 function for Right\n";
    return side2;
}

void Right::SetSide1(float S)
{
    debug << "Entering SetSide1 function for Right\n";
    side = S;
    debug << "Exiting SetSide1 function for Right\n";
}

void Right::SetSide2(float S)
{
    debug << "Entering SetSide2 function for Right\n";
    side2 = S;
    debug << "Exiting SetSide2 function for Right\n";
}

float Right::Perimeter() const
{
    debug << "Entering Perimeter function for Right\n";
    debug << "Exiting Perimeter function for Right\n";
    return side + side2 + sqrt(side*side + side2*side2);
}

float Right::Area() const
{
    debug << "Entering Area function for Right\n";
    debug << "Exiting Area function for Right\n";
    return 0.5 * (side + side2);
}
